packages/iris-grid/src/mousehandlers/IrisGridContextMenuHandler.tsx

```diff
@@ -546,6 +552,47 @@ class IrisGridContextMenuHandler extends GridMouseHandler {
       }
     }
 
+    // Expand/Collapse options
+    if (isExpandableGridModel(model) && model.isRowExpandable(modelRow)) {
+      actions.push({
+        title: model.isRowExpanded(modelRow) ? 'Collapse Row' : 'Expand Row',
+        group: IrisGridContextMenuHandler.GROUP_EXPAND_COLLAPSE,
+        order: 10,
+        action: () => {
+          model.setRowExpanded(modelRow, !model.isRowExpanded(modelRow));
+        },
+      });
+
+      actions.push({
+        title: 'Expand Rows Below',
+        shortcut: SHORTCUTS.TABLE.EXPAND_ROWS_BELOW,
+        group: IrisGridContextMenuHandler.GROUP_EXPAND_COLLAPSE,
+        order: 20,
+        action: () => {
+          model.setRowExpanded(modelRow, true, true);
+        },
+      });
+
+      let expandAll = false;
+      // If there is a row that isn't expanded, expand all from root
+      for (let i = 0; i < rowCount; i += 1) {
```

Comment:
Tables could have billions of rows - we shouldn't iterate through all the rows. 
I think the API could give us the current size of the table and the maximum size of the table when all rows are expanded, then we could just compare the two. But iterating every row we should avoid at all costs. There's got to be a better way to check if there are any expanded rows.

Reply:
Maybe just expose both entries as options. Then you don't need to try to detect if there's expanded rows or not, just have 2 buttons

It might even be preferable since it seems the current way would mean the only way to collapse all is to first expand all. Maybe the user expanded many groups manually and wants to collapse them all again

Reply:
Collapsing the root is not equivalent to collapse all. There may not be a single root node. Also, for tree tables at least the child expanded states are maintained so if you expand the root again, the children are expanded/collapsed as before

Reply:
Is the new expand all behaviour on both rollups and trees, or just rollups?

Reply:
I don't think there's a distinction between the 2 in the JS API. Both are a tree table. Rollup just adds some aggregation columns, but the UI just sees them both as tree tables

Reply:
@emilyhuxng We might need to re-work both the server-side API to support expand multiple roots, and re-consider what options are shown based on the limitations. You can chill on this a bit until we figured it out.

Reply:
As Matt mentioned, if a couple rows are expanded, should the option be Expand All or Collapse All? @dsmmcken

Reply:
If the API is tricky to detect if all are expanded, could just make it based on the target row. So if the row is expanded show "Collapse All". If collapsed show "Expand all"

Reply:
I've updated the parent ticket with requested wording changes.